libdrgn/program.h

```diff
@@ -64,9 +64,18 @@ struct vmcoreinfo {
 	bool pgtable_l5_enabled;
 };
 
+struct drgn_thread {
+    int tid;
+    struct nstring prstatus;
+    struct drgn_object thread;
+};
+
 DEFINE_VECTOR_TYPE(drgn_typep_vector, struct drgn_type *)
 DEFINE_VECTOR_TYPE(drgn_prstatus_vector, struct nstring)
-DEFINE_HASH_MAP_TYPE(drgn_prstatus_map, uint32_t, struct nstring)
+DEFINE_HASH_MAP_TYPE(drgn_thread_map, uint32_t, struct drgn_thread*)
+DEFINE_HASH_MAP_FUNCTIONS(drgn_thread_map, int_key_hash_pair, scalar_key_eq)
```

Comment:
Basically, yes. What I ideally wanted to do was just `typedef struct drgn_thread_set_iterator drgn_thread_iterator`, but I couldn't get that to work for some reason.

Reply:
So does it work to move `struct drgn_thread_iterator { ... };` and the hash map functions to `program.c`? I think it will, since everywhere else just deals with an opaque `struct drgn_thread_iterator *` handle.